components/brave_wallet/browser/bitcoin/bitcoin_rpc.cc

```diff
@@ -0,0 +1,251 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_wallet/browser/bitcoin/bitcoin_rpc.h"
+
+#include "base/command_line.h"
+#include "base/ranges/algorithm.h"
+#include "base/strings/strcat.h"
+#include "base/strings/string_util.h"
+#include "brave/components/brave_wallet/browser/brave_wallet_utils.h"
+#include "brave/components/brave_wallet/common/brave_wallet.mojom-forward.h"
+#include "brave/components/brave_wallet/common/switches.h"
+#include "net/http/http_request_headers.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+
+namespace {
+
+net::NetworkTrafficAnnotationTag GetNetworkTrafficAnnotationTag() {
+  return net::DefineNetworkTrafficAnnotation("bitcoin_rpc", R"(
+      semantics {
+        sender: "Bitcoin RPC"
+        description:
+          "This service is used to communicate with Bitcoin nodes "
+          "on behalf of the user interacting with the native Brave wallet."
+        trigger:
+          "Triggered by uses of the native Brave wallet."
+        data:
+          "Bitcoin JSON RPC response bodies."
+        destination: WEBSITE
+      }
+      policy {
+        cookies_allowed: NO
+        setting:
+          "You can enable or disable this feature on chrome://flags."
+        policy_exception_justification:
+          "Not implemented."
+      }
+    )");
+}
+
+bool IsAsciiAlphaNumeric(const std::string& str) {
+  return base::ranges::all_of(str, &base::IsAsciiAlphaNumeric<char>);
+}
+
+const GURL MakeGetChainHeightUrl(const GURL& base_url) {
+  GURL::Replacements replacements;
+  const std::string path = base::StrCat({base_url.path(), "blocks/tip/height"});
```

Comment:
Is the base_url path guaranteed to end with a `/`? If not, it's probably safer to use `"/blocks/tip/height"`